src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopMetricsParser.cs

```diff
@@ -0,0 +1,163 @@
+
+        /// <summary>
+        /// Column Values in the Hadoop output.
+        /// </summary>
+        private IList<string> columnNames = new List<string> { "Name", "Measurement" };
+
+        /// <summary>
+        /// Split Column Values to find the unit in the Hadoop output.
+        /// </summary>
+        private IList<string> splitColumnNames = new List<string> { "Value", "Unit" };
+
+        /// <summary>
+        /// Constructor for <see cref="HadoopMetricsParser"/>
+        /// </summary>
+        /// <param name="rawText">Raw text to parse.</param>
+        public HadoopMetricsParser(string rawText)
+            : base(rawText)
+        {
+        }
+
+        /// <summary>
+        /// File System Counters result.
+        /// </summary>
+        public DataTable FileSystemCounters { get; set; }
+
+        /// <summary>
+        /// Job Counters result.
+        /// </summary>
+        public DataTable JobCounters { get; set; }
+
+        /// <summary>
+        /// Map Reduce Framework result.
+        /// </summary>
+        public DataTable MapReduceFrameworkCounters { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public List<Metric> Metrics { get; set; }
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            this.Preprocess();
+            this.Sections = TextParsingExtensions.Sectionize(this.PreprocessedText, HadoopMetricsParser.HadoopSectionDelimiter);
+            this.ParseFileSystemCounters();
+            this.ParseJobCounters();
+            this.ParseMapReduceFrameworkCounters();
+
+            this.Metrics =
+            [
+                .. this.FileSystemCounters.GetMetrics(nameIndex: 0, unitIndex: 3, valueIndex: 2, metricRelativity: MetricRelativity.LowerIsBetter),
+                .. this.JobCounters.GetMetrics(nameIndex: 0, unitIndex: 3, valueIndex: 2, metricRelativity: MetricRelativity.LowerIsBetter),
+                .. this.MapReduceFrameworkCounters.GetMetrics(nameIndex: 0, unitIndex: 3, valueIndex: 2, metricRelativity: MetricRelativity.HigherIsBetter),
+            ];
+
+            return this.Metrics;
+        }
+
+        /// <inheritdoc/>
+        protected override void Preprocess()
+        {
+            // Remove all the rows that don't have column sign.
+            List<string> result = new List<string>();
+
+            this.PreprocessedText = this.RawText.Replace("File System Counters", $"{Environment.NewLine}File System Counters");
+            this.PreprocessedText = this.PreprocessedText.Replace("Job Counters", $"{Environment.NewLine}Job Counters");
+            this.PreprocessedText = this.PreprocessedText.Replace("Map-Reduce Framework", $"{Environment.NewLine}Map-Reduce Framework");
+
+            this.PreprocessedText = Regex.Replace(this.PreprocessedText, @"org\.apache\.hadoop\.examples\.terasort\.TeraGen\$Counters[ \r]*\n", string.Empty);
```

Comment:
Why do you need to rename so many things in preprocessing? Maybe you could just Regex catch as is, and then do the rename in the metric names

Reply:
These were few of the output lines with no metric values but more like a title. So, in order to remove it, I had replaced it with empty strings. I am open to other optimized ways to achieve the results.